### PR TITLE
add read timeout

### PR DESCRIPTION
**Pull request by svalaskevicius**
Adds read timeout for socket reads.

A reason to add it within the library rather than timing out skunk calls - it also applies for the healthcheck queries.

### CODE REVIEW

**Comment by svalaskevicius** [comment]
@mpilquist yeah, I noticed that commit, and found no viable alternative means to recover the removed functionality - thus the PR :)

I've rebased on main now, and added a test - please can you have a look?

File: docker-compose.yml [hunk 1]
```diff
@@ -2,18 +2,20 @@ version: '3'
...
+      context: world
     command: -c ssl=on -c ssl_cert_file=/var/lib/postgresql/server.crt -c ssl_key_file=/var/lib/postgresql/server.key
-    volumes:
-      - ./world/world.sql:/docker-entrypoint-initdb.d/world.sql
-      - ./world/server.crt:/var/lib/postgresql/server.crt
-      - ./world/server.key:/var/lib/postgresql/server.key
+    # volumes:
+    #   - ./world/world.sql:/docker-entrypoint-initdb.d/world.sql
+    #   - ./world/server.crt:/var/lib/postgresql/server.crt:ro
+    #   - ./world/server.key:/var/lib/postgresql/server.key:ro
```
Comment (svalaskevicius) [inline docker-compose.yml:12]: I wasn't able to run like this due to mounted key file ownership / permissions within the container - how is it run by other people?

Comment (armanbilge) [reply docker-compose.yml:12]: CI does this.
https://github.com/tpolecat/skunk/blob/f67aceec5119e7bf7a9fc3151bfb310bbfee3e54/.github/workflows/ci.yml#L69-L75

Comment (svalaskevicius) [reply docker-compose.yml:12]: yeah I did this in the Dockerfile, so I suppose this change should be acceptable? ie.. Dockerfile in `world/` doesn't do more than that :)

**Pull request merged**